File: network-equipment-pms/pms-file/src/main/java/com/dp/plat/file/entity/Attachment.java
package com.dp.plat.file.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.dp.plat.common.entity.BaseEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 文件附件实体。
 *
 * <p>统一管理各业务模块（PUNCH_LIST/RMA/DELIVERABLE/IMPL_PROGRESS/ACCEPTANCE 等）的附件，
 * 记录文件元数据、存储位置、上传人、MD5 校验，以及图片类附件的 GPS EXIF 信息和围栏比对结果。</p>
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("pms_attachment")
public class Attachment extends BaseEntity {

    /** 业务类型（PUNCH_LIST/RMA/DELIVERABLE/IMPL_PROGRESS/ACCEPTANCE 等）。 */
    private String bizType;

    /** 业务对象 id。 */
    private Long bizId;

    /** 文件名。 */
    private String fileName;

    /** 文件大小（字节）。 */
    private Long fileSize;

    /** MIME 类型。 */
    private String mimeType;

    /** 上传人 id。 */
    private Long uploadUserId;

    /** 上传人姓名。 */
    private String uploadUserName;

    /** 上传时间。 */
    private LocalDateTime uploadTime;

    /** 文件 MD5 摘要。 */
    private String md5;

    /** 存储路径（本地相对路径或对象存储 key）。 */
    private String storagePath;

    /** 存储类型（LOCAL/OSS/MINIO）。 */
    private String storageType;

    /** GPS 纬度（精度 10,7）。 */
    private BigDecimal gpsLatitude;

    /** GPS 经度（精度 10,7）。 */
    private BigDecimal gpsLongitude;

    /** 照片拍摄时间（来自 EXIF）。 */
    private LocalDateTime photoTakenAt;

    /** GPS 围栏比对结果（NORMAL/ABNORMAL）。 */
    private String geoFenceStatus;
}

File: network-equipment-pms/pms-frontend/src/api/attachment.ts
import { del, get } from '@/utils/request'

/**
 * 附件 API。对应后端 `FileController`，挂载在 `/api/file` 下。
 *
 * <p>统一管理各业务模块（PUNCH_LIST/RMA/DELIVERABLE/IMPL_PROGRESS/ACCEPTANCE/TASK 等）
 * 的附件上传/下载/删除/查询。</p>
 *
 * <p>上传由 `FileUploader` 组件直接调用 `POST /api/file/upload`（multipart/form-data），
 * 本模块仅封装查询、删除、下载地址生成等辅助 API。</p>
 */

/** 附件元数据，与后端 `Attachment` 实体对齐 */
export interface Attachment {
  id?: number
  /** 业务类型（PUNCH_LIST/RMA/DELIVERABLE/IMPL_PROGRESS/ACCEPTANCE/TASK 等） */
  bizType?: string
  /** 业务对象 id（新建场景可为空） */
  bizId?: number
  fileName?: string
  fileSize?: number
  mimeType?: string
  uploadUserId?: number
  uploadUserName?: string
  uploadTime?: string
  md5?: string
  storagePath?: string
  storageType?: string
  geoFenceStatus?: string
}

/** 按业务类型 + 业务 id 查询附件列表 */
export function listAttachmentsByBiz(bizType: string, bizId: number): Promise<Attachment[]> {
  return get<Attachment[]>('/api/file/biz', { bizType, bizId })
}

/** 删除附件（先删存储再删记录） */
export function deleteAttachment(id: number): Promise<boolean> {
  return del<boolean>(`/api/file/${id}`)
}

/** 附件下载地址 */
export function attachmentDownloadUrl(id: number): string {
  return `/api/file/${id}/download`
}

/** 附件缩略图地址（仅图片类附件有效） */
export function attachmentThumbnailUrl(id: number, width = 200, height = 200): string {
  return `/api/file/${id}/thumbnail?width=${width}&height=${height}`
}

/** 格式化文件大小 */
export function formatFileSize(bytes?: number): string {
  if (!bytes || bytes <= 0) return '-'
  if (bytes < 1024) return `${bytes} B`
  if (bytes < 1024 * 1024) return `${(bytes / 1024).toFixed(1)} KB`
  return `${(bytes / (1024 * 1024)).toFixed(2)} MB`
}

/** 格式化上传时间：去 T 截到秒 */
export function formatUploadTime(s?: string): string {
  return s ? s.replace('T', ' ').slice(0, 19) : '-'
}
